lsp-mode.el

```diff
@@ -6093,55 +6110,62 @@ deserialization.")
   (let ((body-received 0)
         leftovers body-length body chunk)
     (lambda (_proc input)
-      (setf chunk (concat leftovers (encode-coding-string input 'utf-8 'nocopy)))
-      (while (not (equal chunk ""))
+      (setf chunk (if (s-blank? leftovers)
+                      input
+                    (concat leftovers input)))
+      (while (not (s-blank? chunk))
         (if (not body-length)
             ;; Read headers
             (if-let ((body-sep-pos (string-match-p "\r\n\r\n" chunk)))
                 ;; We've got all the headers, handle them all at once:
                 (setf body-length (lsp--get-body-length
                                    (mapcar #'lsp--parse-header
                                            (split-string
-                                            (substring chunk
+                                            (substring-no-properties chunk
                                                        (or (string-match-p "Content-Length" chunk)
                                                            (error "Unable to find Content-Length header."))
                                                        body-sep-pos)
                                             "\r\n")))
                       body-received 0
                       leftovers nil
-                      chunk (substring chunk (+ body-sep-pos 4)))
+                      chunk (substring-no-properties chunk (+ body-sep-pos 4)))
 
               ;; Haven't found the end of the headers yet. Save everything
               ;; for when the next chunk arrives and await further input.
               (setf leftovers chunk
-                    chunk ""))
+                    chunk nil))
           (let* ((chunk-length (string-bytes chunk))
                  (left-to-receive (- body-length body-received))
                  (this-body (if (< left-to-receive chunk-length)
-                                (prog1 (substring chunk 0 left-to-receive)
-                                  (setf chunk (substring chunk left-to-receive)))
+                                (prog1 (substring-no-properties chunk 0 left-to-receive)
+                                  (setf chunk (substring-no-properties chunk left-to-receive)))
                               (prog1 chunk
-                                (setf chunk ""))))
+                                (setf chunk nil))))
                  (body-bytes (string-bytes this-body)))
             (push this-body body)
             (setf body-received (+ body-received body-bytes))
             (when (>= chunk-length left-to-receive)
-              (let ((lsp-parsed-message (decode-coding-string
```

Comment:
@yyoncho @ericdallo Since the `lsp-parsed-message` is removed from here, the fix for #1807 will likely be broken.